app/controllers/passwords_controller.rb

```diff
@@ -46,6 +73,11 @@ def reset_params
   end
 
   def validate_confirmation_token
+    if params[:token].nil?
+      redirect_to root_path, alert: t("failure_when_forbidden")
+      return
+    end
```

Comment:
Is there a reason why this clause is added to this and email confirmations? I'm assuming L82 would encapsulate this case.